### PR TITLE
Upload to s3 feature added into interactive workflow

### PR DESCRIPTION
**Pull request by Sahiler**
Moved away from using snowflake for the interactive workflow example as the output can often be unstructured text. 

#### Two main reasons:

1. This made it difficult to create a snowpark sql query dynamically. 
2. Uploading to s3 would be the valid approach to house any unstructured results and add onto the aws flavored demos. 

#### Next steps
I have a `generate_suggested_file_name` task that is currently a WIP, where currently the file name that is used is static in the rest of the workflow. I plan on debugging this flaky task, but currently the script is valid and demo-able. Additionally, will point the .deploy command to the prefect-demos repo main branch once the updated flow code gets merged in.

### CODE REVIEW

**Comment by masonmenges** [comment]
@Sahiler Make sure you run the pre-commits in you're local environment, i.e. pip install pre-commit, then  pre-commit run, I'll get that added to the requirements file for the demos repo so this triggers automatically when you do a git commit

File: flows/aws/interactive-workflow/README.md
Comment (masonmenges) [inline flows/aws/interactive-workflow/README.md:21]: This should probably be 3.8 since that's the lowest version I think we support now

File: flows/aws/interactive-workflow/deploy-interactive-workflow.py
Comment (masonmenges) [inline flows/aws/interactive-workflow/deploy-interactive-workflow.py:4]: We'll want to probably set this up in the dev branch for this repo first to do some fine tuning and then create/test the deployment here https://app.prefect.cloud/account/0ff44498-d380-4d7b-bd68-9b52da03823f/workspace/f579e720-7969-4ab8-93b7-2dfa784903e6/dashboard to start with before we merge into main

File: flows/aws/interactive-workflow/interactive-workflow.py
Comment (masonmenges) [inline flows/aws/interactive-workflow/interactive-workflow.py:58]: Is this needed?

**Review by masonmenges — COMMENTED** [review]
Make sure the pre-commits are run, we'll probably hold off on merging this to main I do have a Dev branch that we'll probably want to merge this to first for testing in the se-dev workspace, still need to make sure the rules for the dev workspace are setup correctly though

File: flows/aws/interactive-workflow/README.md
Comment (EmilRex) [inline flows/aws/interactive-workflow/README.md:51]: It would be helpful to either link to the `S3Bucket` and Marvin docs, or mention the important bits here.

File: flows/aws/interactive-workflow/deploy-interactive-workflow.py
Comment (EmilRex) [inline flows/aws/interactive-workflow/deploy-interactive-workflow.py:1]: Should this file just be named `deploy.py`? That follows the convention set in the datalake example.

File: flows/aws/interactive-workflow/interactive-workflow.py
Comment (EmilRex) [inline flows/aws/interactive-workflow/interactive-workflow.py:26]: Class names should follow the [CapWords convention](https://peps.python.org/pep-0008/#class-names).

File: flows/aws/interactive-workflow/README.md
Comment (EmilRex) [inline flows/aws/interactive-workflow/README.md:26]: "Installation" and "Running the Demo" could be consolidated into a single "Setup" section.

File: flows/aws/interactive-workflow/README.md
Comment (EmilRex) [reply flows/aws/interactive-workflow/README.md:21]: We should probably assume that the user here knows that this requires Python. Do we even need a pre-requisites section?

File: flows/aws/interactive-workflow/README.md
Comment (EmilRex) [inline flows/aws/interactive-workflow/README.md:79]: This doesn't make sense to me. It would probably be better to include a top level doc string in each file with a short description. For example:
```py
"""Utilities for interacting with LLMs via Marvin"""
```

File: flows/aws/interactive-workflow/README.md
Comment (EmilRex) [inline flows/aws/interactive-workflow/README.md:82]: Contribution guidelines should just be at the top level of the repo.

File: flows/aws/interactive-workflow/marvin_extension.py
Comment (EmilRex) [inline flows/aws/interactive-workflow/marvin_extension.py:97]: We should just ask the user to provide it or default to some deterministic naming pattern (UUID or datetime).

File: flows/aws/interactive-workflow/marvin_extension.py
Comment (EmilRex) [inline flows/aws/interactive-workflow/marvin_extension.py:117]: You could simplify this with `write_path` ([ref](https://prefecthq.github.io/prefect-aws/s3/#prefect_aws.s3.S3Bucket.write_path))

File: flows/aws/interactive-workflow/marvin_extension.py
Comment (EmilRex) [inline flows/aws/interactive-workflow/marvin_extension.py:35]: Why is the data getting passed as a `JSON` block?

**Review by EmilRex — CHANGES_REQUESTED** [review]
Left some comments.

One bigger point is that reading over the README, it's not clear what this example is trying to accomplish. I understand what features are being used, but not what is happening, nor why I would use this pattern.

File: flows/aws/interactive-workflow/README.md
Comment (masonmenges) [reply flows/aws/interactive-workflow/README.md:21]: That's probably fair, I think in general we should assume relevant information for these demos outside of perfect specific functionality is familiar to whoever might look at it

File: flows/aws/interactive-workflow/deploy-interactive-workflow.py
Comment (Sahiler) [reply flows/aws/interactive-workflow/deploy-interactive-workflow.py:1]: Good idea! I'll change the naming convention so its more consistent

File: flows/aws/interactive-workflow/marvin_extension.py
Comment (Sahiler) [reply flows/aws/interactive-workflow/marvin_extension.py:117]: Ahh great idea, I'll get that added in instead

File: flows/aws/interactive-workflow/marvin_extension.py
Comment (Sahiler) [reply flows/aws/interactive-workflow/marvin_extension.py:35]: I wanted to use the JSON block to showcase some 'data lineage' on the query results. Plus there are events thrown each time this block is used/updated, so any trigger based work can build off of that

File: flows/aws/interactive-workflow/interactive-workflow.py
Comment (Sahiler) [reply flows/aws/interactive-workflow/interactive-workflow.py:59]: Looks like one of my debug statements snuck in, I'll get rid of that. Thank you for catching!

File: flows/aws/interactive-workflow/marvin_extension.py
Comment (masonmenges) [reply flows/aws/interactive-workflow/marvin_extension.py:35]: I think it might make more sense to enjoy a custom event in that case but for the purposes of this demo I don't think that's necessary.

**Comment by Sahiler** [comment]
Have a few readme updates and new s3 function to add in, converting to a draft in the meantime

**Pull request closed without merging**
